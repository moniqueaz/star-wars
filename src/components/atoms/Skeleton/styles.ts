import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${( { theme } ) => css`
    width: 100%;
    height: 100%;
    background: ${theme.colors.gray};
    border-radius: ${theme.border.radius};
    overflow: hidden;
    animation: fullView 0.5s forwards cubic-bezier(0.25, 0.46, 0.45, 0.94);
    
    @keyframes fullView {
      100% {
        width: 100%;
      }
    }
  `}
`;

export const Animate = styled.div`
  ${( { theme } ) => css`
    width: 100%;
    height: 100%;
    background: ${theme.colors.gray};
    opacity: 30%;
    animation: shimmer 2s infinite linear;
    background: linear-gradient(135deg, ${theme.colors.gray} 30%, ${theme.colors.white} 50%, ${theme.colors.gray} 70%);
    background-size: 300% 100%;
    @keyframes shimmer {
      0% {
        background-position: 100% 0;
      }
      100% {
        background-position: 0% 0;
      }
    }
  `}
`;
