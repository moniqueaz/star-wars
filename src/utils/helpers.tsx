export const fetcher = (url: string) => fetch(url).then(res => res.json());

export const getRandomInteger = (min: number, max: number) => {
  return Math.floor(Math.random() * (max - min)) + min;
};
