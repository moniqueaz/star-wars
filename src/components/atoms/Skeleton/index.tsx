import * as S from './styles';

const Skeleton = () => (
  <S.Wrapper>
    <S.Animate />
  </S.Wrapper>
);

export default Skeleton;
