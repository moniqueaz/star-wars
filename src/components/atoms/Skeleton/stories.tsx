import { Story, Meta } from '@storybook/react/types-6-0';
import Skeleton from '.';

export default {
  title: 'Atoms/Skeleton',
  component: Skeleton,
} as Meta;

export const Default: Story = () => {
  return <div style={{ width: '200px', height: '100px' }}>
    <Skeleton />
  </div>;
};
