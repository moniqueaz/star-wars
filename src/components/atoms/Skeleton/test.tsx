import { renderWithTheme } from 'utils/tests/helpers';

import Skeleton from '.';

describe('<Skeleton />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Skeleton />);
    expect(container.querySelector('div')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  } );
} );
