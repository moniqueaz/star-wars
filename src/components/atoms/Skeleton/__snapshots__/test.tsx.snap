// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> should render the heading 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background: #8F8F8F;
  border-radius: 0.4rem;
  overflow: hidden;
  -webkit-animation: fullView 0.5s forwards cubic-bezier(0.25,0.46,0.45,0.94);
  animation: fullView 0.5s forwards cubic-bezier(0.25,0.46,0.45,0.94);
}

.c1 {
  width: 100%;
  height: 100%;
  background: #8F8F8F;
  opacity: 30%;
  -webkit-animation: shimmer 2s infinite linear;
  animation: shimmer 2s infinite linear;
  background: linear-gradient(135deg,#8F8F8F 30%,#f1f1f1 50%,#8F8F8F 70%);
  background-size: 300% 100%;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
</div>
`;
